tests/test_dbapi2.py

```diff
@@ -856,3 +857,61 @@ def test_interface_error_reading_result_set_after_commits():
 ])
 def test_finding_operation(statement, operation):
     assert _sql_operation(statement) == operation
+
+
+@pytest.mark.parametrize(
+    "values",
+    [
+        [5, 10],
+        ("hello", "hi"),
+        [0.25, 0.35, 0.25],
+        (b"123", b"456"),
+    ],
+)
+def test_parameter_binding_arrays(values):
+    # GIVEN
+    conn = connect("mattdb", "dev")
+    cursor = conn.cursor()
+
+    # WHEN
+    cursor.execute("select * from carray(%(values)s)", dict(values=values))
+    results = cursor.fetchall()
+
+    # THEN
+    assert results == [[v] for v in values]
+    conn.close()
+
+
+@pytest.mark.parametrize(
+    "values,exc_msg",
+    [
+        (
+            [],
+            "Can't bind list value [] for parameter 'values': "
+            + "ValueError: empty lists cannot be bound",
+        ),
+        (
+            (),
+            "Can't bind tuple value () for parameter 'values': "
+            + "ValueError: empty tuples cannot be bound",
+        ),
+        (
+            [1, "hello"],
+            "Can't bind list value [1, 'hello'] for parameter 'values': "
+            + "ValueError: all list elements must be the same type",
+        ),
+        (
+            (1j,),
+            "Can't bind tuple value (1j,) for parameter 'values': "
+            + "ValueError: Cannot bind a tuple of complex",
+        ),
```

Comment:
I would add another test case for nested lists or nested tuples, just to check that the error message makes sense for them too.

Reply:
It'll just say "Cannot bind a tuple of tuple" (since we only go one level deep), but sure, that's an easy test to add.